[11.5.1] Assertion '_M_stddev > _RealType(0)' failed

When using noise tag in my ray sensor such as
```
      <sensor name='ouster_lidar' type='ray'>    
          <visualize>0</visualize>    
          <update_rate>20</update_rate>    
          <ray>    
            <scan>    
              <vertical>    
                <samples>32</samples>    
                <resolution>1</resolution>    
                <min_angle>-0.289725</min_angle>    
                <max_angle>0.289725</max_angle>    
              </vertical>    
              <horizontal>    
                <samples>512</samples>    
                <resolution>1</resolution>    
                <min_angle>-3.14159</min_angle>    
                <max_angle>3.14159</max_angle>    
              </horizontal>    
            </scan>    
            <range>    
              <min>0.8</min>    
              <max>100</max>    
              <resolution>0.03</resolution>    
            </range>    
            <noise>      
            </noise>    
          </ray>    
        </sensor>    
```
Gazebo throws this error 
```/
usr/include/c++/11.1.0/bits/random.h:1992: std::normal_distribution<_RealType>::param_type::param_type(_RealType, _RealType) [with _RealType = double]: Assertion '_M_stddev > _RealType(0)' failed.
```

Still having the issue, here is what I have for now when compiling with https://github.com/bombela/backward-cpp (really great project btw):

```
/usr/include/c++/11.1.0/bits/random.h:1992: std::normal_distribution<_RealType>::param_type::param_type(_RealType, _RealType) [with _RealType = double]: Assertion '_M_stddev > _RealType(0)' failed.
Stack trace (most recent call last) in thread 49085:
#23   Object "", at 0xffffffffffffffff, in 
#22   Object "/usr/lib/libc.so.6", at 0x7f0a6d9505e2, in clone
#21   Object "/usr/lib/libpthread.so.0", at 0x7f0a6e0a3258, in 
#20   Source "/build/gcc/src/gcc/libstdc++-v3/src/c++11/thread.cc:82:18", in  [0x7f0a6dc503c3]
#19   Object "/usr/lib/libgazebo_physics.so.11", at 0x7f0a6d2a793d, in gazebo::physics::World::RunLoop()
#18   Object "/usr/lib/libgazebo_physics.so.11", at 0x7f0a6d2a4e50, in gazebo::physics::World::Step()
#17   Object "/usr/lib/libgazebo_physics.so.11", at 0x7f0a6d2a430c, in gazebo::physics::World::ProcessMessages()
#16   Object "/usr/lib/libgazebo_physics.so.11", at 0x7f0a6d299f5f, in gazebo::physics::World::ProcessFactoryMsgs()
#15   Object "/usr/lib/libgazebo_physics.so.11", at 0x7f0a6d296e53, in gazebo::physics::World::LoadModel(std::shared_ptr<sdf::v9::Element>, boost::shared_ptr<gazebo::physics::Base>)
#14   Object "/usr/lib/libgazebo_physics.so.11", at 0x7f0a6d238c88, in gazebo::physics::Model::Load(std::shared_ptr<sdf::v9::Element>)
#13   Object "/usr/lib/libgazebo_physics.so.11", at 0x7f0a6d22c8f5, in gazebo::physics::Model::LoadLinks()
#12   Object "/usr/lib/libgazebo_physics.so.11", at 0x7f0a6d1165b8, in gazebo::physics::ODELink::Load(std::shared_ptr<sdf::v9::Element>)
#11   Object "/usr/lib/libgazebo_physics.so.11", at 0x7f0a6d2143e1, in gazebo::physics::Link::Load(std::shared_ptr<sdf::v9::Element>)
#10   Object "/usr/lib/libgazebo_sensors.so.11", at 0x7f0a6d7d15cd, in 
#9    Object "/usr/lib/libgazebo_sensors.so.11", at 0x7f0a6d7cb7bb, in gazebo::sensors::SensorManager::OnCreateSensor(std::shared_ptr<sdf::v9::Element>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int)
#8    Object "/usr/lib/libgazebo_sensors.so.11", at 0x7f0a6d7cb193, in gazebo::sensors::SensorManager::CreateSensor(std::shared_ptr<sdf::v9::Element>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int)
#7    Object "/usr/lib/libgazebo_sensors.so.11", at 0x7f0a6d7b7822, in gazebo::sensors::RaySensor::Load(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
#6    Object "/usr/lib/libgazebo_sensors.so.11", at 0x7f0a6d7b23f0, in gazebo::sensors::NoiseFactory::NewNoiseModel(std::shared_ptr<sdf::v9::Element>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
#5    Object "/usr/lib/libgazebo_sensors.so.11", at 0x7f0a6d78f8f2, in gazebo::sensors::GaussianNoiseModel::Load(std::shared_ptr<sdf::v9::Element>)
#4    Object "/usr/lib/libgazebo_sensors.so.11", at 0x7f0a6d78eabd, in gazebo::sensors::GaussianNoiseModel::SampleBias()
#3    Object "/usr/lib/libignition-math6.so.6", at 0x7f0a6c664c56, in ignition::math::v6::Rand::DblNormal(double, double)
#2    Object "/usr/lib/libignition-math6.so.6", at 0x7f0a6c64ca09, in
#1    Object "/usr/lib/libc.so.6", at 0x7f0a6d878861, in abort
#0    Object "/usr/lib/libc.so.6", at 0x7f0a6d88ed22, in gsignal
Aborted (Signal sent by tkill() 48949 1000)
[ERROR] [1632146010.176045, 0.426000]: Spawn service failed. Exiting.
/opt/ros/noetic/lib/gazebo_ros/gzserver: line 41: 48949 Aborted                 (core dumped) GAZEBO_MASTER_URI="$desired_master_uri" GAZEBO_MODEL_DATABASE_URI="$desired_model_database_uri" gzserver $final
[gazebo-2] process has died [pid 48884, exit code 134, cmd /opt/ros/noetic/lib/gazebo_ros/gzserver -e ode /opt/ros/noetic/share/turtlebot3_gazebo/worlds/empty.world __name:=gazebo __log:=/home/nihil/.ros/log/1a2231de-1a1a-11ec-a200-68071562f7af/gazebo-2.log].
log file: /home/nihil/.ros/log/1a2231de-1a1a-11ec-a200-68071562f7af/gazebo-2*.log
[spawn_urdf-4] process has died [pid 48892, exit code 1, cmd /opt/ros/noetic/lib/gazebo_ros/spawn_model -urdf -model turtlebot3_burger -x 0.0 -y 0.0 -z 0.0 -param robot_description __name:=spawn_urdf __log:=/home/nihil/.ros/log/1a2231de-1a1a-11ec-a200-68071562f7af/spawn_urdf-4.log].
log file: /home/nihil/.ros/log/1a2231de-1a1a-11ec-a200-68071562f7af/spawn_urdf-4*.log
```

ok, here is a more detailed backtrace:

```
process[gazebo-2]: started with pid [63823]
process[gazebo_gui-3]: started with pid [63828]
process[spawn_urdf-4]: started with pid [63834]
[ INFO] [1632148762.384886961]: Finished loading Gazebo ROS API Plugin.
[ INFO] [1632148762.385952178]: waitForService: Service [/gazebo/set_physics_properties] has not been advertised, waiting...
[ INFO] [1632148762.461495390]: Finished loading Gazebo ROS API Plugin.
[ INFO] [1632148762.462915168]: waitForService: Service [/gazebo_gui/set_physics_properties] has not been advertised, waiting...
[ INFO] [1632148762.818747795]: waitForService: Service [/gazebo/set_physics_properties] is now available.
[ INFO] [1632148762.847993368, 0.005000000]: Physics dynamic reconfigure ready.
/usr/include/c++/11.1.0/bits/random.h:1992: std::normal_distribution<_RealType>::param_type::param_type(_RealType, _RealType) [with _RealType = double]: Assertion '_M_stddev > _RealType(0)' failed.
Stack trace (most recent call last) in thread 136111:
#23   Object "", at 0xffffffffffffffff, in 
#22   Object "/usr/lib/libc.so.6", at 0x7f1ead7d85e2, in clone
#21   Object "/usr/lib/libpthread.so.0", at 0x7f1eadf2b258, in 
#20   Source "/build/gcc/src/gcc/libstdc++-v3/src/c++11/thread.cc:82:18", in  [0x7f1eadad83c3]
#19   Source "/usr/src/debug/gazebo-11.7.0/gazebo/physics/World.cc:563:7", in  [0x7f1ead12e8ad]
        560:         (!this->dataPtr->stopIterations ||
        561:          (this->dataPtr->iterations < this->dataPtr->stopIterations));)
        562:     {
      > 563:       this->Step();
        564:     }
        565:   }
        566:   else
#18   Source "/usr/src/debug/gazebo-11.7.0/gazebo/physics/World.cc:779:3", in  [0x7f1ead12bdc0]
        777:   gazebo::util::IntrospectionManager::Instance()->NotifyUpdates();
        778: 
      > 779:   this->ProcessMessages();
        780: 
        781:   DIAG_TIMER_STOP("World::Step");
#17   Source "/usr/src/debug/gazebo-11.7.0/gazebo/physics/World.cc:2861:5", in  [0x7f1ead12b27c]
       2858:     this->ProcessPlaybackControlMsgs();
       2859:     this->ProcessEntityMsgs();
       2860:     this->ProcessRequestMsgs();
      >2861:     this->ProcessFactoryMsgs();
       2862:     this->ProcessModelMsgs();
       2863:     this->ProcessLightFactoryMsgs();
       2864:     this->ProcessLightModifyMsgs();
#16   Source "/usr/include/c++/11.1.0/ext/atomicity.h:92:12", in  [0x7f1ead120e9f]
         89:   inline void
         90:   __attribute__((__always_inline__))
         91:   __atomic_add_single(_Atomic_word* __mem, int __val)
      >  92:   { *__mem += __val; }
         93: 
         94:   inline _Atomic_word
         95:   __attribute__ ((__always_inline__))
#15   Source "/usr/include/c++/11.1.0/bits/shared_ptr_base.h:709:26", in  [0x7f1ead11dd93]
        706:       : _M_pi(__r._M_pi)
        707:       {
        708:    if (_M_pi != nullptr)
      > 709:      _M_pi->_M_add_ref_copy();
        710:       }
        711: 
        712:       __shared_count&
#14   Source "/usr/src/debug/gazebo-11.7.0/gazebo/physics/Model.cc:154:3", in  [0x7f1ead0bfbb8]
        151:   if (this->sdf->HasElement("allow_auto_disable"))
        152:     this->SetAutoDisable(this->sdf->Get<bool>("allow_auto_disable"));
        153: 
      > 154:   this->LoadLinks();
        155: 
        156:   this->LoadModels();
#13   Source "/usr/include/c++/11.1.0/ext/atomicity.h:92:12", in  [0x7f1ead0b3815]
         89:   inline void
         90:   __attribute__((__always_inline__))
         91:   __atomic_add_single(_Atomic_word* __mem, int __val)
      >  92:   { *__mem += __val; }
         93: 
         94:   inline _Atomic_word
         95:   __attribute__ ((__always_inline__))
#12   Source "/usr/include/c++/11.1.0/ext/atomicity.h:92:12", in  [0x7f1eacf9d5a8]
         89:   inline void
         90:   __attribute__((__always_inline__))
         91:   __atomic_add_single(_Atomic_word* __mem, int __val)
      >  92:   { *__mem += __val; }
         93: 
         94:   inline _Atomic_word
         95:   __attribute__ ((__always_inline__))
#11   Source "/usr/include/c++/11.1.0/bits/std_function.h:560:9", in  [0x7f1ead09b301]
        557:       {
        558:    if (_M_empty())
        559:      __throw_bad_function_call();
      > 560:    return _M_invoker(_M_functor, std::forward<_ArgTypes>(__args)...);
        561:       }
        562: 
        563: #if __cpp_rtti
#10   Source "/usr/include/c++/11.1.0/bits/shared_ptr_base.h:732:8", in std::_Function_handler<void(std::shared_ptr<sdf::v9::Element>, const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, unsigned int), std::_Bind<void (gazebo::sensors::SensorManager::*(gazebo::sensors::SensorManager*, std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>))(std::shared_ptr<sdf::v9::Element>, const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, unsigned int)> >::_M_invoke(_Any_data &__functor, shared_ptr<sdf::v9::Element> &&__args#0, basic_string<char, std::char_traits<char>, std::allocator<char> > &__args#1, basic_string<char, std::char_traits<char>, std::allocator<char> > &__args#2, unsigned int &&__args#3) [0x7f1ead65a4dd]
        729:       {
        730:    _Sp_counted_base<_Lp>* __tmp = __r._M_pi;
        731:    __r._M_pi = _M_pi;
      > 732:    _M_pi = __tmp;
        733:       }
        734: 
        735:       long
#9    Source "/usr/src/debug/gazebo-11.7.0/gazebo/sensors/SensorManager.cc:484:21", in  [0x7f1ead6546cb]
        481:     const std::string &_parentName,
        482:     const uint32_t _parentId)
        483: {
      > 484:   this->CreateSensor(_elem, _worldName, _parentName, _parentId);
        485: }
        486: 
        487: //////////////////////////////////////////////////
#8    Source "/usr/include/c++/11.1.0/ext/atomicity.h:92:12", in  [0x7f1ead6540a3]
         89:   inline void
         90:   __attribute__((__always_inline__))
         91:   __atomic_add_single(_Atomic_word* __mem, int __val)
      >  92:   { *__mem += __val; }
         93: 
         94:   inline _Atomic_word
         95:   __attribute__ ((__always_inline__))
#7    Source "/usr/src/debug/gazebo-11.7.0/gazebo/sensors/RaySensor.cc:112:36", in  [0x7f1ead640732]
        109:   if (rayElem->HasElement("noise"))
        110:   {
        111:     this->noises[RAY_NOISE] =
      > 112:         NoiseFactory::NewNoiseModel(rayElem->GetElement("noise"),
        113:         this->Type());
        114:   }
#6    Source "/usr/include/c++/11.1.0/ext/atomicity.h:92:12", in  [0x7f1ead63b300]
         89:   inline void
         90:   __attribute__((__always_inline__))
         91:   __atomic_add_single(_Atomic_word* __mem, int __val)
      >  92:   { *__mem += __val; }
         93: 
         94:   inline _Atomic_word
         95:   __attribute__ ((__always_inline__))
#5    Source "/usr/include/c++/11.1.0/bits/basic_string.h:150:14", in  [0x7f1ead6187d2]
        147: #endif
        148: 
        149:       // Use empty-base optimization: http://www.cantrip.org/emptyopt.html
      > 150:       struct _Alloc_hider : allocator_type // TODO check __is_final
        151:       {
        152: #if __cplusplus < 201103L
        153:    _Alloc_hider(pointer __dat, const _Alloc& __a = _Alloc())
#4    Source "/usr/src/debug/gazebo-11.7.0/gazebo/sensors/GaussianNoiseModel.cc:223:38", in gazebo::sensors::GaussianNoiseModel::SampleBias() [0x7f1ead61799d]
        220: void GaussianNoiseModel::SampleBias()
        221: {
        222:   this->bias =
      > 223:       ignition::math::Rand::DblNormal(this->biasMean, this->biasStdDev);
        224:   // With equal probability, we pick a negative bias (by convention,
        225:   // rateBiasMean should be positive, though it would work fine if
        226:   // negative).
#3  | Source "/usr/include/c++/11.1.0/bits/regex_error.h:57:33", in (result_type __mean, result_type __stddev)
    |    55:       _S_error_brack,
    |    56:       _S_error_paren,
    | >  57:       _S_error_brace,
    |    58:       _S_error_badbrace,
    |    59:       _S_error_range,
    | Source "/usr/src/debug/ign-math-ignition-math6_6.8.0/src/Rand.cc:2027:9", in (double __mean, double __stddev)
    | Source "/usr/src/debug/ign-math-ignition-math6_6.8.0/src/Rand.cc:1989:2", in (double __mean, double __stddev)
      Source "/usr/include/c++/11.1.0/bits/random.h:1989:2", in double ignition::math::v6::Rand::DblNormal(double _mean, double _sigma) [0x7f1eac4d1c87]
       1986:    param_type() : param_type(0.0) { }
       1987: 
       1988:    explicit
      >1989:    param_type(_RealType __mean, _RealType __stddev = _RealType(1))
       1990:    : _M_mean(__mean), _M_stddev(__stddev)
       1991:    {
       1992:      __glibcxx_assert(_M_stddev > _RealType(0));
#2    Source "/usr/include/c++/11.1.0/x86_64-pc-linux-gnu/bits/c++config.h:504:5", in std::__replacement_assert(char *__file, int __line, char *__function, char *__condition) [0x7f1eac4adfa9]
        501:   {
        502:     __builtin_printf("%s:%d: %s: Assertion '%s' failed.\n", __file, __line,
        503:                 __function, __condition);
      > 504:     __builtin_abort();
        505:   }
        506: }
        507: #define __glibcxx_assert_impl(_Condition)                         \
#1    Object "/usr/lib/libc.so.6", at 0x7f1ead700861, in abort
#0    Object "/usr/lib/libc.so.6", at 0x7f1ead716d22, in gsignal
Aborted (Signal sent by tkill() 74837 1000)
[ERROR] [1632152440.753201, 0.194000]: Spawn service failed. Exiting.
/opt/ros/noetic/lib/gazebo_ros/gzserver: line 41: 74837 Aborted                 (core dumped) GAZEBO_MASTER_URI="$desired_master_uri" GAZEBO_MODEL_DATABASE_URI="$desired_model_database_uri" gzserver $final
[gazebo-2] process has died [pid 74758, exit code 134, cmd /opt/ros/noetic/lib/gazebo_ros/gzserver -e ode /opt/ros/noetic/share/turtlebot3_gazebo/worlds/empty.world __name:=gazebo __log:=/home/nihil/.ros/log/06930ec2-1a29-11ec-a19d-73d58ec4b5d2/gazebo-2.log].
log file: /home/nihil/.ros/log/06930ec2-1a29-11ec-a19d-73d58ec4b5d2/gazebo-2*.log
[spawn_urdf-4] process has died [pid 74765, exit code 1, cmd /opt/ros/noetic/lib/gazebo_ros/spawn_model -urdf -model turtlebot3_burger -x 0.0 -y 0.0 -z 0.0 -param robot_description __name:=spawn_urdf __log:=/home/nihil/.ros/log/06930ec2-1a29-11ec-a19d-73d58ec4b5d2/spawn_urdf-4.log].
log file: /home/nihil/.ros/log/06930ec2-1a29-11ec-a19d-73d58ec4b5d2/spawn_urdf-4*.log
```

--- a/gazebo/sensors/GaussianNoiseModel.cc
+++ b/gazebo/sensors/GaussianNoiseModel.cc
@@ -219,13 +219,16 @@ double GaussianNoiseModel::GetBias() const
 //////////////////////////////////////////////////
 void GaussianNoiseModel::SampleBias()
 {
-  this->bias =
-      ignition::math::Rand::DblNormal(this->biasMean, this->biasStdDev);
-  // With equal probability, we pick a negative bias (by convention,
-  // rateBiasMean should be positive, though it would work fine if
-  // negative).
-  if (ignition::math::Rand::DblUniform() < 0.5)
-    this->bias = -this->bias;
+  if(this->biasStdDev)
+  {
+    this->bias =
+        ignition::math::Rand::DblNormal(this->biasMean, this->biasStdDev);
+    // With equal probability, we pick a negative bias (by convention,
+    // rateBiasMean should be positive, though it would work fine if
+    // negative).
+    if (ignition::math::Rand::DblUniform() < 0.5)
+      this->bias = -this->bias;
+  }
 }
 
 //////////////////////////////////////////////////

